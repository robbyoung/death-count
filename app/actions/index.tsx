export * from './actionTypes';
export * from './addPlaythrough';
export * from './addDeath';
export * from './addGame';
export * from './addDeathDetail';
export * from './completeDeath';