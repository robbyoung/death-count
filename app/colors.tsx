export const backgroundColor = '#592a29';
export const buttonColor = '#e84641';
export const white = '#ffffff';