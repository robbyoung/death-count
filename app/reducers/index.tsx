import characters from './characters';
import deaths from './deaths';
import { combineReducers } from 'redux';

export default combineReducers({
    characters,
    deaths,
});
