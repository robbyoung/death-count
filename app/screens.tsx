export enum Screens {
    Home = 'Home',
    NewDeath = 'NewDeath',
    GamesList = 'GamesList',
}
