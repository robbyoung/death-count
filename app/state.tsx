export interface Game {
    id: string;
    name: string;
    options: string;
}

export interface DeathOptions {
    name: string;
    options: string[];
}

export interface Playthrough {
    id: string;
    name: string;
    description: string;
    gameId: string;
}

export interface Death {
    id: string;
    playthroughId: string;
}

export interface State {
    games: Game[];
    playthroughs: Playthrough[];
    deaths: Death[];
    currentPlaythrough: string;
    currentGame: string;
}