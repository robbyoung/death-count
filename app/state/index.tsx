export * from './state';
export * from './testState';