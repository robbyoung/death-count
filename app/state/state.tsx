export interface Game {
    id: string;
    name: string;
    selected: boolean;
    complete: boolean;
}

export interface Playthrough {
    id: string;
    name: string;
    gameId: string;
    selected: boolean;
}

export interface Death {
    id: string;
    playthroughId: string;
    complete: boolean;
    details: { [id: string] : string };
}

export interface OptionSet {
    id: string;
    gameId: string;
    title: string;
    options: string[];
    selected: boolean;
}

export interface State {
    games: Game[];
    playthroughs: Playthrough[];
    deaths: Death[];
    optionSets: OptionSet[];
}