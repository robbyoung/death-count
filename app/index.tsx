import { createAppContainer } from 'react-navigation';
import { createStackNavigator } from 'react-navigation-stack';
import { Screens } from './screens';
import Home from './containers/home';
import NewDeath from './containers/newDeath';
import GamesList from './containers/gamesList';
import NewGame from './containers/newGame';
import Loading from './containers/loading';
import Startup from './containers/startup';
import PlaythroughsList from './containers/playthroughsList';

const MainNavigator = createStackNavigator(
	{
		Loading: { screen: Loading },
		Home: { screen: Home },
		NewDeath: { screen: NewDeath },
		GamesList: { screen: GamesList },
		NewGame: { screen: NewGame },
		Startup: { screen: Startup },
		PlaythroughsList: { screen: PlaythroughsList }
	},
	{
		initialRouteName: Screens.Loading,
	},
);

export default createAppContainer(MainNavigator);