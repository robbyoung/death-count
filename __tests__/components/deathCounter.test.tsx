import React from 'react';
import renderer from 'react-test-renderer';
import DeathCounter from '../../app/components/deathCounter';
import { createTestState } from '../../app/state';

describe('Death Counter Component', () => {
    it('renders correctly with no deaths', () => {
        const component = renderer.create(<DeathCounter
            deaths={[]}
            onPress={() => undefined}
        />);
        expect(component.toJSON()).toMatchSnapshot();
    });

    it('renders correctly with multiple deaths', () => {
        const state = createTestState(1, 1, 10, 0, 0);
        const component = renderer.create(<DeathCounter
            deaths={state.deaths}
            onPress={() => undefined}
        />);
        expect(component.toJSON()).toMatchSnapshot();
    })
});