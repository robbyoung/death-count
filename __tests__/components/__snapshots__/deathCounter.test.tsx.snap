// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Death Counter Component renders correctly with multiple deaths 1`] = `
<View
  style={
    Object {
      "alignContent": "center",
      "flexDirection": "row",
      "justifyContent": "center",
      "marginTop": 30,
    }
  }
>
  <View
    accessible={true}
    clickable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#e84641",
        "borderRadius": 25,
        "color": "#ffffff",
        "margin": 5,
        "opacity": 1,
        "padding": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "borderRadius": 100,
          "color": "#ffffff",
          "fontSize": 30,
          "textAlign": "center",
          "textAlignVertical": "center",
        }
      }
    >
      10
    </Text>
  </View>
  <Text
    style={
      Object {
        "borderRadius": 100,
        "color": "#ffffff",
        "fontSize": 30,
        "textAlign": "center",
        "textAlignVertical": "center",
      }
    }
  >
    Deaths So Far
  </Text>
</View>
`;

exports[`Death Counter Component renders correctly with no deaths 1`] = `
<View
  style={
    Object {
      "alignContent": "center",
      "flexDirection": "row",
      "justifyContent": "center",
      "marginTop": 30,
    }
  }
>
  <View
    accessible={true}
    clickable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#e84641",
        "borderRadius": 25,
        "color": "#ffffff",
        "margin": 5,
        "opacity": 1,
        "padding": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "borderRadius": 100,
          "color": "#ffffff",
          "fontSize": 30,
          "textAlign": "center",
          "textAlignVertical": "center",
        }
      }
    >
      0
    </Text>
  </View>
  <Text
    style={
      Object {
        "borderRadius": 100,
        "color": "#ffffff",
        "fontSize": 30,
        "textAlign": "center",
        "textAlignVertical": "center",
      }
    }
  >
    Deaths So Far
  </Text>
</View>
`;
